_posts/2023-12-11-augmented-reality-in-travel-applications.md

```diff
@@ -0,0 +1,130 @@
+
+## The setup
+
+Experiments have a great advantage: you can start a greenfield project. An option that is not always available to developers. In my case, I opted for the following simple setup:
+
+- A simple HTML file containing AR.js markup. We will go into this in more detail in the next section.
+- A JSON: two representative locations to the left and right of my apartment at the time with geo-coordinates coming from our database with over 173,000 POIs: [Augsburg City Hall](https://www.holidaycheck.de/pi/rathaus/24e3daee-7218-3ff1-8111-613f11d0dbc4) and the [Fuggerei](https://www.holidaycheck.de/pi/fuggerei/7a9ada69-b008-31af-8361-0e12f2d62158). What a great way to get to know my home town!
+- A JavaScript function: To load nearby places and enrich them with some data.
+- A simple express server: To deliver the HTML. Why? Because I don't want to run around with my laptop. The app should be accessible from my iPhone.
+
+## Initialize the camera in the browser
+
+For an executable version, you only need markup - no code (except for the AR.js library). The location-based camera needs a scene to which it must be added to work. It looks like this:
+
+```html
+<a-scene
+  raycaster="objects: [gps-entity-place];"
+  arjs="sourceType: webcam;"
+  embedded
+>
+  <a-camera gps-camera rotation-reader></a-camera>
+</a-scene>
+```
+
+Let's take a look at the attributes:
+
+- `raycaster`: We want to interact with places. To do this, we need to render 3D objects. Under the hood, [AR.js uses Three.js](https://ar-js-org.github.io/AR.js-Docs/ui-events/) for this.
+- `arjs`: Sure, we have to initialize the scene. We also specify `sourceType: webcam` to ensure that the smartphone camera is also used.
+
+For `a-camera` we need `gps-camera` to activate [Location AR](https://ar-js-org.github.io/AR.js-Docs/location-based#camera-component-gps-new-camera-gps-projected-camera-or-gps-camera) and `rotation-reader` to actively track the position when the camera is rotated. This way, we know where the user is pointing their device.
+
+The app was now ready to be tested for the first time via the smartphone browser. To do this, I simply started the server and called up the IP of my machine. Once access to the camera has been granted, a red camera appears in the address bar:
+
+<figure>
+    <img src="img/posts/2023-12-11-augmented-reality-in-travel-applications/iphone-status-bar-while-recording.png" alt="iPhone status bar while recording" class="centered" />
+    <figcaption>The status bar of my iPhone while recording in Safari.</figcaption>
+</figure>
+
+## Interact with remote locations
+
+First of all, we need our own location. To get our current geo-position, we use the [Navigation API](https://developer.mozilla.org/en-US/docs/Web/API/Navigation_API). We now use the callback from `getCurrentPosition` for our further logic:
```

Comment:
From singular "... I simply started", you go to the plural "... we need".
IMHO - pick one, and stick with it.

Reply:
Sticked to "we" now.